compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -479,6 +479,9 @@ generateSrcFromLf env = noLoc mod
 
         convConDetails :: LF.Type -> Gen (HsConDeclDetails GhcPs)
         convConDetails = \case
+            -- empty variant constructor (see issue #7207)
+            LF.TUnit ->
```

Comment:
This means that a variant where I explicilty specify a Unit argument will also be decompiled to a Prefix con right? I think that’s perfectly reasonable but probably worth documenting in a comment and perhaps warning on in the preprocessor.

Reply:
Yes, that's what we decided on in the issue and I'm preparing the warning in a separate PR.